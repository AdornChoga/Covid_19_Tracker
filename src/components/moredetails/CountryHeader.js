import React from 'react';
import PropTypes from 'prop-types';
import classes from './MoreDetails.module.css';

const CountryHeader = (props) => {
  const { country } = props;
  const {
    name, image, todayOpenCases,
  } = country;
  return (
    <div className={classes.country}>
      <img src={image} alt="map" />
      <div>
        <span className={classes.country_name}>{name}</span>
        <span className={classes.country_open_cases}>
          Open Cases:&nbsp;&nbsp;&nbsp;
          {todayOpenCases}
        </span>
      </div>
    </div>
  );
};

CountryHeader.propTypes = {
  country: PropTypes.shape({
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    todayOpenCases: PropTypes.number.isRequired,
  }).isRequired,
};

export default CountryHeader;
