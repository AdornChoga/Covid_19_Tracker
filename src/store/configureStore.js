import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import statistics from './statistics/statistics';

const reducer = combineReducers({
  statistics,
});

const store = createStore(reducer, applyMiddleware(logger, thunk));

export default store;
