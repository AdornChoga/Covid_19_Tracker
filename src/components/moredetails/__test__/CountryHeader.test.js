import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../../store/configureStore';
import CountryHeader from '../CountryHeader';

it('should create country header component snapshot', () => {
  const country = {
    name: 'Canada',
    image: 'image.png',
    todayOpenCases: 4949,
  };
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <CountryHeader country={country} />
      </Router>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});
