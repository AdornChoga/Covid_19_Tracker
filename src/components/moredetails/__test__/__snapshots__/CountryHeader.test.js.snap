// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create country header component snapshot 1`] = `
<div
  className="country"
>
  <img
    alt="map"
    src="image.png"
  />
  <div>
    <span
      className="country_name"
    >
      Canada
    </span>
    <span
      className="country_open_cases"
    >
      Open Cases:   
      4949
    </span>
  </div>
</div>
`;
